docs/mutability.md

```diff
@@ -0,0 +1,63 @@
+# Mutability in Substratus
+
+## Models
+
+## Datasets
+
+## Servers
+
+```yaml
+kind: Server
+
+spec:
+
+  params: {} # Change to field triggers redeploy.
+
+  image:
```

Comment:
the API spec makes me believe image is now a string and there no longer is image.name. The doc on mutability probably needs to be updated to match latest API design?

Reply:
good eye on this one 👁️